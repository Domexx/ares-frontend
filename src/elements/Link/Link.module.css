/* === === === === ../elements/Link === === === === */

.Link {
  margin-top: 0;
}
