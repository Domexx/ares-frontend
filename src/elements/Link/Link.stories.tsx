import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import LinkComponent, { Props } from './Link';

const Component = {
  title: 'Elements/Link',
  component: LinkComponent,
  argTypes: {
    isExternal: { control: { type: 'boolean' } },
  },
};

const Template: Story<ComponentProps<typeof LinkComponent>> = (args) => {
  return <LinkComponent {...args} />;
};

export const Link = Template.bind({});
export default Component;

Link.args = {
  url: '/test',
  text: 'What happened to virgil',
  isExternal: true,
} as Props;
