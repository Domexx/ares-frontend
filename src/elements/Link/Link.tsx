/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import NextLink from 'next/link';
import { useRouter } from 'next/router';
import s from './Link.module.css';

export type Props = {
  url: string;
  text: string;
  locale: boolean;
  isExternal?: boolean;
};

const Link: React.FC<Props> = ({ url, text, locale, isExternal = false }) => {
  const router = useRouter();

  // @NOTE: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#security_and_privacy
  const externalProps = {
    rel: 'noreferrer noopener',
    target: '_blank',
  };

  const localeToPass = locale ? router.locale : false;

  return (
    <NextLink locale={localeToPass} href={url} passHref>
      <a {...(isExternal && externalProps)} className={s['text']}>
        {text}
      </a>
    </NextLink>
  );
};

export default Link;
