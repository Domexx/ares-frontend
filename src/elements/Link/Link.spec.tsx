import React from 'react';

import { render } from '@testing-library/react';

import Link from './Link';

describe('Link Component', () => {
  const props = {
    url: 'https://url.com',
    text: 'Virgil',
  };

  it('should render properly as a relative link', () => {
    const { container } = render(<Link {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render properly as an external link', () => {
    const { container } = render(<Link {...props} isExternal />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
