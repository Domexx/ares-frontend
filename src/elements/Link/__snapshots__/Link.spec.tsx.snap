// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component should render properly as a relative link 1`] = `
<a
  class="text"
  href="https://url.com"
>
  Virgil
</a>
`;

exports[`Link Component should render properly as an external link 1`] = `
<a
  class="text"
  href="https://url.com"
  rel="noreferrer noopener"
  target="_blank"
>
  Virgil
</a>
`;
